target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-deco-m5-v2.dts

```diff
@@ -0,0 +1,371 @@
+// SPDX-License-Identifier: ISC
+// Copyright (c) 2015, The Linux Foundation. All rights reserved.
+// Copyright (c) 2021, Frank Veltmans <veltmans.frank@gmail.com>
+ 
+#include "qcom-ipq4019.dtsi"
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/soc/qcom,tcsr.h>
+#include <dt-bindings/leds/common.h>
+
+ / {
+	model = "TP-Link Deco M5";
+	compatible = "tplink,deco-m5", "qcom,ipq4019";
+
+	chosen {
+		bootargs-append = " root=/dev/mtdblock16";
+	};
+
+	aliases {
+		spi0 = "/soc/spi@78b5000";
+		spi1 = "/soc/spi@78b6000";
+		i2c0 = "/soc/i2c@78b7000";
+		i2c1 = "/soc/i2c@78b8000";
+		led-boot = &led_blue;
+		led-failsafe = &led_red;
+		led-running = &led_green;
+		led-upgrade = &led_red;
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&tlmm 18 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+
+	soc {
+		interrupt-controller@b000000 {
+			compatible = "qcom,msm-qgic2";
+			interrupt-controller;
+			#interrupt-cells = < 0x03 >;
+			reg = < 0xb000000 0x1000 0xb002000 0x1000 >;
+			linux,phandle = < 0x01 >;
+			phandle = < 0x01 >;
+		};
+
+		pinctrl@1000000 {
+			serial_0_pins: serial_0_pinmux {
+				pinmux {
+					pins = "gpio16", "gpio17";
+					function = "blsp_uart0";
+					bias-disable;
+				};
+			};
+
+			serial_1_pins: serial_1_pinmux {
+				pinmux {
+					pins = "gpio8", "gpio9", "gpio10", "gpio11";
+					function = "blsp_uart1";
+					bias-disable;
+				};
+			};
+
+			spi_0_pins: spi_0_pinmux {
+				pinmux {
+					pins = "gpio13", "gpio14", "gpio15";
+					function = "blsp_spi0";
+				};
+				pinmux_cs {
+					pins = "gpio12";
+					function = "gpio";
+				};
+				pinconf {
+					pins = "gpio13", "gpio14", "gpio15";
+					drive-strength = <12>;
+					bias-disable;
+				};
+				pinconf_cs {
+					pins = "gpio12";
+					drive-strength = <2>;
+					bias-disable;
+					output-high;
+				};
+			};
+
+			i2c_0_pins: i2c_0_pinmux {
+				pinmux {
+					pins = "gpio20", "gpio21";
+					function = "blsp_i2c0";
+				};
+				pinconf {
+					pins = "gpio20", "gpio21";
+					drive-strength = <16>;
+					bias-disable;
+				};
+			};
+
+			wifi_pins: wifi_pinmux {
+				pinmux_1 {
+					pins = "gpio37";
+					function = "gpio";
+					drive-strength = <6>;
+					bias-pull-up;
+					output-high;
+				};
+
+				pinmux_2 {
+					pins = "gpio42";
+					function = "gpio";
+					drive-strength = <6>;
+					bias-pull-up;
+					output-high;
+				};
+
+				pinmux_3 {
+					pins = "gpio43";
+					function = "gpio";
+					drive-strength = <6>;
+					bias-pull-up;
+				};
+			};
+		};
+
+		blsp_dma: dma@7884000 {
+			status = "okay";
+		};
+
+		spi_0: spi@78b5000 {
+			pinctrl-0 = <&spi_0_pins>;
+			pinctrl-names = "default";
+			status = "okay";
+			cs-gpios = <&tlmm 12 GPIO_ACTIVE_HIGH>;
+
+			m25p80@0 {
+				#address-cells = <1>;
+				#size-cells = <0>;
+				reg = <0>;
+				compatible = "jedec,spi-nor";
+				spi-max-frequency = <24000000>;
```

Comment:
What's the SPI-NOR model used?
Usually, they can go way above 24MHz, and 50MHz is the SPI controller max